LanguageFeatures/Inline-classes/static_analysis_inline_class_A12_t02.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let DV be an inline class declaration named Inline with type
+/// parameters X1 extends B1, .. Xs extends Bs. Assume that DV declares a final
+/// instance variable with name id and type R.
+///
+/// We say that the declared representation type of Inline is R, and the
+/// instantiated representation type corresponding to Inline<T1,.. Ts> is
+/// [T1/X1, .. Ts/Xs]R.
+/// ...
+/// Let V be an inline type of the form Inline<T1, .. Ts>, and let R be the
+/// corresponding instantiated representation type. If R is non-nullable then V
+/// is a proper subtype of Object, and V is non-nullable. Otherwise, V is a
+/// proper subtype of Object?, and V is potentially nullable.
+///
+/// @description Checks that if an instantiated representation type `R` is
+/// non-nullable then it is a compile-time error to assign `Object` to `V`
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+inline class V1 {
+  final int? id;
+  V1(this.id);
+}
+
+inline class V2<T1, T2 extends num?> {
+  final num? id;
```

Comment:
I think it would be helpful to use a representation type where the phrase 'instantiated representation type' means something.

That is, the representation type should contain one or more type variables, such that it makes a difference which actual type arguments we're passing.

So, for instance, the representation type could be `T1`.

We should then consider `V2<int, .../*this one doesn't matter, let's say Never*/...>`. The point is that the instantiated representation type is `int`, and that is a non-nullable type.

Then it should still be an error to assign `Object()` to a variable of type `V2<int, ...>`.

There may be a different test doing this, but the other side of that coin is that it _is_ an error to assign an expression of type `V2<int?, ...>` to a variable of type `Object`.